components/ILIAS/OpenIdConnect/classes/class.ilOpenIdConnectUserSync.php

```diff
@@ -109,17 +117,17 @@ protected function transformToXml(): void
             $this->writer->xmlElement('Login', [], $this->int_account);
         }
 
-        $this->writer->xmlElement('ExternalAccount', [], $this->ext_account);
-        $this->writer->xmlElement('AuthMode', ['type' => self::AUTH_MODE], null);
+        $this->writer->xmlElement('ExternalAccount', array(), $this->ext_account);
+        $this->writer->xmlElement('AuthMode', array('type' => self::AUTH_MODE), null);
 
         $this->parseRoleAssignments();
 
         if ($this->needsCreation()) {
-            $this->writer->xmlElement('Active', [], "true");
-            $this->writer->xmlElement('TimeLimitOwner', [], 7);
-            $this->writer->xmlElement('TimeLimitUnlimited', [], 1);
-            $this->writer->xmlElement('TimeLimitFrom', [], time());
-            $this->writer->xmlElement('TimeLimitUntil', [], time());
+            $this->writer->xmlElement('Active', array(), "true");
+            $this->writer->xmlElement('TimeLimitOwner', array(), 7);
+            $this->writer->xmlElement('TimeLimitUnlimited', array(), 1);
+            $this->writer->xmlElement('TimeLimitFrom', array(), time());
+            $this->writer->xmlElement('TimeLimitUntil', array(), time());
         }
 
         foreach ($this->settings->getProfileMappingFields() as $field => $lng_key) {
```

Comment:
This will not work for "Customer User Fields", they are simply not part of this field map.